msm/skill_entry.py

```diff
@@ -114,6 +114,13 @@ def __init__(self, name, path, url='', sha='', msm=None):
         self.url = url
         self.sha = sha
         self.msm = msm
+        if msm:
+            u = url.lower()
+            u = u[:-4] if u.endswith(".git") else u
```

Comment:
Suggestion: `u = u[:-len(".git")] if ...`.

Reply:
Also, why don't we put this next to the `url.rstrip` above so that it normalizes the whole attribute? (cloning etc. should still work without `.git`, right?)